effectful-core/src/Effectful/Internal/Monad.hs

```diff
@@ -294,14 +310,31 @@ instance IOE :> es => MonadBaseControl IO (Eff es) where
 data Prim :: Effect where
   Prim :: Prim m r
 
+type instance DispatchOf Prim = 'Static
+
 -- | Run an 'Eff' computation with primitive state-transformer actions.
 runPrim :: IOE :> es => Eff (Prim : es) a -> Eff es a
-runPrim = evalData (DataA Prim)
+runPrim = evalData (DataR Prim)
 
 instance Prim :> es => PrimMonad (Eff es) where
   type PrimState (Eff es) = RealWorld
   primitive = unsafeEff_ . IO
 
+----------------------------------------
+-- Dispatch
+
+-- | A dispatch type. For more information consult the documentation in
+-- "Effectful.Dispatch.Dynamic" and "Effectful.Dispatch.Static".
+data Dispatch = Dynamic | Static
+
+-- | A dispatch type of a particular effect.
+type family DispatchOf (e :: Effect) :: Dispatch
```

Comment:
This is fine as-is, but I have an idea of how we could avoid the need for this altogether: https://github.com/arybczak/effectful/pull/48#issuecomment-1006702437

Reply:
I don't think you can avoid this, unfortunately. E.g. if you do:

```haskell
newtype DataR :: Effect -> Type where
  DataR :: { unDataR :: StaticRep e } -> DataR e

data family StaticRep (e :: Effect) :: Type
```

and then

```haskell
data State s :: Effect

newtype instance StaticRep (State s) = State s
```

Then you can't use `send`, because there is nothing to send, but you can still mix static and dynamic effects in weird ways.

As for distinguishing static effects from dynamic by having a public wrapper for them, I experimented with it a long time ago and it really clutters type signatures.